clock/test_clock_test.go

```diff
@@ -61,3 +62,20 @@ func TestTickAfter(t *testing.T) {
 	tickOrTimeOut(ticker2, true)
 	tickOrTimeOut(ticker3, false)
 }
+
+func TestTickAfterConcurrent(t *testing.T) {
+	ch := make(chan struct{})
+	c := NewTestClockWithTickSignal(testTime, ch)
+
+	go func() {
+		select {
+		case <-c.TickAfter(time.Second):
```

Comment:
Doesn't the test now already finish before this can be executed? I'd expect a wait for the goroutine

Reply:
Good catch! Fixed.